app/routes.py

```diff
@@ -1,2 +1,187 @@
-from flask import Blueprint
+from app import db
+from app.models.task import Task
+from flask import Blueprint, jsonify, make_response, request
+from sqlalchemy import asc, desc
+from datetime import datetime, date
+from app.models.goal import Goal
+import requests, json
+from dotenv import load_dotenv
+import os
+from app.models.utility_func import *
+
+load_dotenv()
+
+
+task_bp = Blueprint("task", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goal", __name__, url_prefix="/goals")
+
+@task_bp.route("", methods=["GET"])
+def get_tasks():
+    all_tasks = Task.query.all()
+    if not all_tasks:
+        return jsonify([]), 200
+
+    sort_query = request.args.get("sort", default="asc") 
+    if sort_query == "desc":
+        all_tasks = Task.query.order_by(desc(Task.title)).all()
+    else:
+        all_tasks = Task.query.order_by(asc(Task.title)).all()
+
+    tasks_response = [task.convert_a_task_to_dict()for task in all_tasks]
+    return jsonify(tasks_response), 200
+
+@task_bp.route("", methods=["POST"])
+def post_a_task():
+    request_body = request.get_json()
+
+    try:  
+        if type(request_body["completed_at"]) == str:
+            datetime_completed_at = from_str_to_datetime(request_body["completed_at"])
+        else:
+            datetime_completed_at = request_body["completed_at"]
+
+        new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=datetime_completed_at)
+        
+        db.session.add(new_task)
+        db.session.commit()
+
+        return new_task.concate_task_key_to_a_dict_with_return_code(201)
+    except KeyError:
+        return keyError_message()
+
+@task_bp.route("/<int:task_id>", methods=["GET"])
+def get_a_task_with_id(task_id):
+    task = Task.query.get_or_404(task_id)
```

Comment:
I see that you are using `get_or_404` here .... just some information on using it .... this generates a text/html response by default rather than JSON. In a production setting, you would want to
research how to customize the default flask error handling to return a JSON result, or
research how to use a decorator to provide lookup error handling that can be shared across routes.